src/requests/consumption.js

```diff
@@ -0,0 +1,744 @@
+ * @returns {Object} - Response contains task object
+ */
+const assignSequenceNumbers = (tasks) => {
+	let sequenceCounter = 1
+	return tasks.map((task) => {
+		if (!task.sequence_no) {
+			task.sequence_no = sequenceCounter++
+		}
+		return task
+	})
+}
+
+const createSolutions = async (resourceDetails, programDetails) => {
+	try {
+		let solutionsToCreate = []
+		resourceDetails.forEach((resource) => {
+			const solutionTemplate = {
+				resourceType: [common.SOLUTIONS_RESOURCE_TYPE[resource.type]],
+				language: resource?.languages ? resource?.languages.map((language) => language.label) : [],
+				keywords: resource?.keywords
+					? Array.isArray(resource?.keywords)
+						? resource?.keywords
+						: resource?.keywords.split(',')
+					: [],
+				concepts: resource?.concepts ? resource?.concepts : [],
+				themes: resource?.themes ? resource?.themes : [],
+				flattenedThemes: resource?.flattenedThemes ? resource?.flattenedThemes : [],
+				entities: resource?.entities ? resource?.entities : [],
+				registry: resource?.registry ? resource?.registry : [],
+				isRubricDriven: resource?.isRubricDriven ? true : false,
+				enableQuestionReadOut: resource?.enableQuestionReadOut ? true : false,
+				captureGpsLocationAtQuestionLevel: resource?.captureGpsLocationAtQuestionLevel ? true : false,
+				isAPrivateProgram: false,
+				allowMultipleAssessemts: resource?.allowMultipleAssessemts ? true : false,
+				isDeleted: false,
+				pageHeading: 'Domains',
+				minNoOfSubmissionsRequired: resource?.minNoOfSubmissionsRequired
+					? resource?.minNoOfSubmissionsRequired
+					: 1,
+				rootOrganisations: resource?.organization
+					? resource?.organization.map((organization) => organization.id)
+					: [],
+				createdFor: resource?.organization ? resource?.organization.map((organization) => organization.id) : [],
+				deleted: false,
+				name: resource?.title,
+				programExternalId: programDetails.externalId,
+				entityType: common.SOLUTIONS_ENTITY_TYPE[resource.type]
+					? common.SOLUTIONS_ENTITY_TYPE[resource.type]
+					: resource?.entityType
+					? resource?.entityType
+					: null,
+				type: common.SOLUTIONS_TYPE[resource.type] ? common.SOLUTIONS_TYPE[resource.type] : null,
+				subType: common.SOLUTIONS_TYPE[resource.type] ? common.SOLUTIONS_TYPE[resource.type] : null,
+				isReusable: false,
+				externalId: utils.generateUniqueId(),
+				programId: programDetails._id,
+				programName: programDetails.name,
+				programDescription: programDetails.description,
+				status: common.STATUS_ACTIVE.toLowerCase(),
+				updatedAt: new Date(),
+				createdAt: new Date(),
+				__v: 0,
+				scope: programDetails.scope,
+				projectTemplateId: null,
+				updatedBy: 1,
+				endDate: programDetails.end_date,
+				startDate: programDetails.start_date,
+			}
+			solutionsToCreate.push(solutionTemplate)
+		})
+
+		const solutionCollection = mongoDb.collection(COLLECTIONS.SOLUTIONS)
+		await solutionCollection.insertMany(solutionsToCreate)
```

Comment:
@adithyadinesh0412  why do we need insertMany here ? can we do single insert ?

Reply:
There can be multiple solutions in programs rollout. So it is an array of solutions getting inserted. @priyanka-TL